docs/source/model_design/designing_vivarium_model/interactive_sim/index.rst

```diff
@@ -65,113 +65,143 @@ Another useful resource is the `Vivarium InteractiveContext documentation <https
-do this, we will use the `InteractiveContext` function in Vivarium. 
+You'll do this by building a model specification object from the default YAML file, and then changing it. Here is an example:
 
-First, you'll need to locate the model specs file. It will be in the engineering repo usually 
-in a file path similar to: :code:`<project_repo_name>/src/<project_repo_name>/model_specifications/`. 
-It will be a .yaml file. 
+.. code:: python
 
-If you can't locate the proper model specs file or are unsure - ask the engineers for help! 
+  import <YOUR_MODEL_NAME>
+  # model_specifications/model_spec.yaml is the typical location within the engineering repo for the
+  # default spec to live. Ask the engineers if you can't find it.
+  path = Path(<YOUR_MODEL_NAME>.__file__.replace('__init__.py', 'model_specifications/model_spec.yaml'))
 
-Once you have that, you can follow this code to create a simulation:
+  from vivarium.framework.configuration import build_model_specification
 
-::
+  custom_model_specification = build_model_specification(path)
 
-  path = Path('<file_path_for_model_spec_file>')
-  
-  sim = InteractiveContext(path, setup=False)
+  # Customizations: will be specific to your model and use case!
+  # Delete observer configuration
+  del custom_model_specification.configuration.observers
+  # Change population size
+  custom_model_specification.configuration.population.population_size = 50_000
 
-This creates an object called `sim` that is the simulation. 
+  # Delete the "metrics" component (an observer)
+  del custom_model_specification.components.vivarium_public_health.metrics
+  # Filter out components with 'Observer' or 'Stratifier' in the name
+  custom_model_specification.components['<YOUR_MODEL_NAME>.components'] = [
+      c for c in custom_model_specification.components['<YOUR_MODEL_NAME>.components']
+      if 'Observer' not in c and 'Stratifier' not in c
+  ]
 
-.. _interactive_setup_3:
+This is essentially like editing the YAML file of the model specification, which you should
+be able to view in the engineering repository for reference.
+If you want to delete an entire section, for example:
+
+.. code:: yaml
+
+  configuration:
+    observers:
+      ...
+
+you use :code:`del custom_model_specification.configuration.observers`.
+You can do this with any section, using :code:`del custom_model_specification.X.Y.Z` where X, Y, and Z
+are the first, second, and third level keys.
+
+If you just want to change a value, you can set it, as we have with the population size in the example.
+
+You can also use the default value to create your new value. That's what's happening with the last lines in
+the example, where we take whatever is in the list at :code:`<YOUR_MODEL_NAME>.components` under :code:`components`
+and remove only the items that have 'Observer' or 'Stratifier' in them.
+This is equivalent to editing a YAML file that looks like this...
```

Comment:
I used to show the differences when I was editing the file. But I think now the code shows what the differences are. If you want to double-check, it might be good to print out the model spec before and after the changes?

Reply:
Ahh I see. I'll add a print of the file after the changes then. I think it helps me to look and see that the coded changes actually happened to confirm the code did what I expected.

Reply:
Not sure if this is what you mean by "the file," but note that this does not actually change the YAML _file_, only the specification object in memory. You can see one representation of that specification by simply running a cell with `model_specification` in it. ConfigTrees tend to give a little TMI for my liking but it should be sufficient to see what is happening.